lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralLinearModelAlgorithm.cxx

```diff
@@ -562,8 +567,9 @@ Scalar GeneralLinearModelAlgorithm::maximizeReducedLogLikelihood()
   OptimizationProblem problem(reducedLogLikelihoodFunction);
   problem.setMinimization(false);
   problem.setBounds(optimizationBounds_);
-  solver_.setStartingPoint(initialParameters);
   solver_.setProblem(problem);
+  if (solver_.getImplementation()->getClassName() != "MultiStart")
+    solver_.setStartingPoint(initialParameters);
```

Comment:
it would be cleaner to use a try/catch block instead of inspecting the implementation

Reply:
Why would it be cleaner?

Reply:
agreed with @jschueller

Reply:
it is more generic: if we add another solver class which doesnt support setSP, we dont have to add its class name in these ifs

Reply:
Done (and also at the other places pointed out by @sofianehaddad).